barretenberg/cpp/pil/avm/main.pil

```diff
@@ -670,8 +606,9 @@ namespace main(256);
     //===== Memory Slice Constraints ============================================
     pol commit sel_slice_gadget; // Selector to activate a slice gadget operation in the gadget (#[PERM_MAIN_SLICE]).
 
+    // FIXME: this fails when retsize is 0
     // Activate only if tag_err is disabled
-    sel_slice_gadget = (sel_op_calldata_copy + sel_op_external_return) * (1 - tag_err);
+    // (1 - tag_err) * (sel_op_calldata_copy + sel_op_external_return - sel_slice_gadget)= 0;
```

Comment:
ib is storing ret_size (for return opcode) and copy_size (for calldatacopy). Therefore, I think we can adapt the relation with:
ib * (1 - tag_err) * (sel_op_calldata_copy + sel_op_external_return - sel_slice_gadget)= 0;

Reply:
Yep, the above relation enforces:
ib != 0 && (sel_op_calldata_copy == 1 || sel_op_external_return == 1) && tag_err == 0 ==> sel_slice_gadget == 1